src/api/spec/models/branch_package_spec.rb

```diff
@@ -31,5 +31,50 @@
         expect(Project.where(name: "#{home_project.name}:branches:BaseDistro:Update")).to exist
       end
     end
+
+    context 'project with ImageTemplates attribute' do
+      let(:attribute_type) { AttribType.find_by_namespace_and_name!('OBS', 'ImageTemplates') }
+
+      context 'auto cleanup attribute' do
+        let(:leap_project) { create(:project, name: 'openSUSE_Leap') }
+        let(:apache) { create(:package, name: 'apache2', project: leap_project) }
+        let!(:image_templates_attrib) { create(:attrib, attrib_type: attribute_type, project: leap_project) }
+        let(:branch_package) { BranchPackage.new(project: leap_project.name, package: apache.name) }
+
+        it 'is set to 14 if there is no default' do
+          branch_package.branch
+          project = Project.find_by_name(user.branch_project_name("openSUSE_Leap"))
+          expect(14.days.from_now - project.attribs.first.values.first.value.to_time).to be < 1.minute
+        end
+
+        it 'is set to the default' do
+          allow(Configuration).to receive(:cleanup_after_days).and_return(42)
+          branch_package.branch
+          project = Project.find_by_name(user.branch_project_name("openSUSE_Leap"))
+          expect(42.days.from_now - project.attribs.first.values.first.value.to_time).to be < 1.minute
+        end
+      end
+    end
+
+    context 'project without ImageTemplates attribute' do
+      let(:leap_project) { create(:project, name: 'openSUSE_Leap') }
+      let(:apache) { create(:package, name: 'apache2', project: leap_project) }
+      let(:branch_package) { BranchPackage.new(project: leap_project.name, package: apache.name) }
+
+      context 'auto cleanup attribute' do
+        it 'is set to the default' do
+          allow(Configuration).to receive(:cleanup_after_days).and_return(42)
+          branch_package.branch
+          project = Project.find_by_name(user.branch_project_name("openSUSE_Leap"))
+          expect(42.days.from_now - project.attribs.first.values.first.value.to_time).to be < 1.minute
```

Comment:
As discussed today, maybe a project.attribs.where better here.
Also maybe we should think about introducing TimeCop gem or something similiar...